deployment/actionManger.go

```diff
@@ -52,7 +66,20 @@ func NewActionManager(activator *activator.Activator, dr *database.Driver, worke
 			}
 
 			mgr.pathHashMux.Lock()
-			mgr.pathHashes[path] = updatedID.Hash
+			mgr.pathHashes[path].hash = updatedID.Hash
+			mgr.pathHashMux.Unlock()
+
+			mgr.NotifyComponentStateChanged()
+		}
+	}()
```

Comment:
add comment: should we be batching hash updates in one lock?

Reply:
I was thinking this as well. The question is like, do we want a thread for each channel. One thread will be updated by the autoscaler periodically. The other will be changing the hash as needed.